spec/integration/inventory_spec.rb

```diff
@@ -279,8 +319,11 @@ module Hawkular::Inventory::RSpec
       resource_path = CanonicalPath.new(feed_id: feed_id, resource_ids: [hawk_escape_id('Local~~')])
       config = @client.get_config_data_for_resource(resource_path)
 
+      # product name is missing
       expect(config['value']['Server State']).to eq('running')
-      expect(config['value']['Product Name']).to eq('Hawkular')
```

Comment:
that's correct. The question is what makes the API, to some extend it's also the data that are being sent, or at least their format. I agree, the tests w/ those 'ifs' look cumbersome

> a current server with the old api

That's pretty artificial, isn't it? Current server is current, because it has the current versions of component with their current APIs :] If you mean current APIs with old data, that would require changing the agent version, which won't work w/o additional effort.